lib/rpmsg/rpmsg_virtio.c

```diff
@@ -610,17 +611,33 @@ int rpmsg_init_vdev(struct rpmsg_virtio_device *rvdev,
 		    rpmsg_ns_bind_cb ns_bind_cb,
 		    struct metal_io_region *shm_io,
 		    struct rpmsg_virtio_shm_pool *shpool)
+{
+	return rpmsg_init_vdev_with_config(rvdev, vdev, ns_bind_cb, shm_io,
+					   shpool, &RPMSG_VIRTIO_DEFAULT_CONFIG);
+}
+
+int rpmsg_init_vdev_with_config(struct rpmsg_virtio_device *rvdev,
+				struct virtio_device *vdev,
+				rpmsg_ns_bind_cb ns_bind_cb,
+				struct metal_io_region *shm_io,
+				struct rpmsg_virtio_shm_pool *shpool,
+				const struct rpmsg_virtio_config *config)
 {
 	struct rpmsg_device *rdev;
 	const char *vq_names[RPMSG_NUM_VRINGS];
 	vq_callback callback[RPMSG_NUM_VRINGS];
 	int status;
 	unsigned int i, role;
 
+	if (config == NULL) {
+		return RPMSG_ERR_PARAM;
+	}
+
```

Comment:
here could be 

```
#ifndef VIRTIO_SLAVE_ONLY
	role = rpmsg_virtio_get_role(rvdev);
	if (role == RPMSG_MASTER) {
		if (config == NULL) {
			return RPMSG_ERR_PARAM;
		}
		rvdev->config = *config;
	}
#endif /*!VIRTIO_SLAVE_ONLY*/

```

Reply:
It's fine for me to update this way, because currently configuration applies only to the MASTER role.

I implemented updating config for both roles in case we extend it in the future with options applicable to both. But the implementation can be generalized later, when such option is created.